### ISSUE TITLE
bad_alloc exception with xenon::find_first

### ISSUE THREAD

**Issue by wythe** [intrig/xenon#59]
On windows, reproduce with:

```c++
  xenon::spec_server d("icd.xddl");
  auto decoded_msg = xenon::parse(d, "0101011107952901a3de0a800982511effffffff");
  auto c = xenon::find_first(decoded_msg, "DATA/MSG_TYPE");
```

Compliments: @rgyger

**Comment by wythe** [comment]
There  was a bug in the xddl definition that is fixed in #60.  The underlying issue still has to be resolved --  the engine should handle gracefully.

**Comment by wythe** [comment]
Underlying issue resovlved in #61
